qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,231 @@
+            offset_guess (float): Optional, an initial guess of the offset
+            t1_bounds (list of two floats): Optional, lower bound and uper bound to T1
+            amplitude_bounds (list of two floats): Optional, lower bound and uper bound to the amplitude
+            offset_bounds (list of two floats): Optional, lower bound and uper bound to the offset
+            kwargs: Trailing unused function parameters
+
+        Returns:
+            The analysis result with the estimated T1
+        """
+
+        circuit_unit = experiment_data._data[0]["metadata"]["unit"]
+        dt_factor_in_sec = experiment_data._data[0]["metadata"].get("dt_factor_in_sec", None)
+        if dt_factor_in_sec is None:
+            dt_factor_in_microsec = 1
+            result_unit = circuit_unit
+        else:
+            dt_factor_in_microsec = dt_factor_in_sec * 1000000
+            result_unit = "us"
+
+        size = len(experiment_data._data)
+        delays = np.zeros(size, dtype=float)
+        means = np.zeros(size, dtype=float)
+        stddevs = np.zeros(size, dtype=float)
+
+        for i, circ in enumerate(experiment_data._data):
+            delays[i] = circ["metadata"]["delay"] * dt_factor_in_microsec
+            count0 = circ["counts"].get("0", 0)
+            count1 = circ["counts"].get("1", 0)
+            shots = count0 + count1
+            means[i] = count1 / shots
+            stddevs[i] = np.sqrt(means[i] * (1 - means[i]) / shots)
+            # problem for the fitter if one of the std points is
+            # exactly zero
+            if stddevs[i] == 0:
+                stddevs[i] = 1e-4
+
+        if t1_guess is None:
+            t1_guess = np.mean(delays)
+        else:
+            t1_guess = t1_guess * dt_factor_in_microsec
+        if offset_guess is None:
+            offset_guess = means[-1]
+        if amplitude_guess is None:
+            amplitude_guess = means[0] - offset_guess
+        if t1_bounds is None:
+            t1_bounds = [0, np.inf]
+        if amplitude_bounds is None:
+            amplitude_bounds = [0, 1]
+        if offset_bounds is None:
+            offset_bounds = [0, 1]
+
+        fit_out, fit_err, fit_cov, chisq = curve_fit_wrapper(
+            exp_fit_fun,
+            delays,
+            means,
+            stddevs,
+            p0=[amplitude_guess, t1_guess, offset_guess],
+        )
+
+        analysis_result = AnalysisResult(
+            {
+                "value": fit_out[1],
+                "stderr": fit_err[1],
+                "unit": result_unit,
+                "label": "T1",
+                "fit": {
+                    "params": fit_out,
+                    "stderr": fit_err,
+                    "labels": ["amplitude", "T1", "offset"],
+                    "chisq": chisq,
+                    "cov": fit_cov,
+                },
+                "quality": self._fit_quality(fit_out, fit_err, chisq),
+            }
+        )
+
+        return analysis_result, None
```

Comment:
You might consider returning the function you fit the curve to as well here -- if the experiment analyzers consistently do that it'll make it easier for users to plot the fits than digging through your code to find the name you selected for the function , and also will future-proof you against future analyzers that might have multiple fit functions to pick between.

Reply:
Daniel asked a similar thing for the curve fit PR but im not sure how would we serialize a generic python function for storing in result DB?

One thing that isn't in this PR but needs to be added later is that we were planning to have fit plots automatically be generated by the the analysis class

Reply:
I guess I was just imagining it in the return value, but I guess the analysis classes could pull the fit function in as a class member, and then return the name of that member